src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -471,24 +462,21 @@ private bool PartStartsWith(string candidate, TextSpan candidatePart, string pat
 
         private PatternMatchKind? TryAllLowerCamelCaseMatch(
             string candidate,
-            ArrayBuilder<TextSpan> candidateHumps,
-            TextChunk patternChunk,
+            in TemporaryArray<TextSpan> candidateHumps,
+            in TextChunk patternChunk,
             out ImmutableArray<TextSpan> matchedSpans)
         {
-            var matcher = new AllLowerCamelCaseMatcher(
-                _includeMatchedSpans, candidate, candidateHumps, patternChunk, _textInfo);
-            return matcher.TryMatch(out matchedSpans);
+            var matcher = new AllLowerCamelCaseMatcher(_includeMatchedSpans, candidate, patternChunk.Text, _textInfo);
+            return matcher.TryMatch(candidateHumps, out matchedSpans);
         }
 
         private PatternMatchKind? TryUpperCaseCamelCaseMatch(
             string candidate,
-            ArrayBuilder<TextSpan> candidateHumps,
-            TextChunk patternChunk,
+            in TemporaryArray<TextSpan> candidateHumps,
+            in TextChunk patternChunk,
             CompareOptions compareOption,
             out ImmutableArray<TextSpan> matchedSpans)
         {
-            var patternHumps = patternChunk.PatternHumps;
```

Comment:
💡 This method could have close to its original form by changing this line to use a `ref readonly var`:

```csharp
ref readonly var patternHumps = ref patternChunk.PatternHumps;
```

Reply:
nice.  thanks.